### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by FrankFeng** [lh3/minimap2#47]
Hello,

I encountered an "core dumped" error when mapping PacBio Sequel long reads (human whole genome data) against the hg38 reference. The causative read was isolated and can be downloaded at https://www.dropbox.com/s/w3lqeskl621af70/culprit.fa?dl=0

The command I used is `minimap2 -t 1 -L -ax map-pb hg38_no_alt.fasta.pacbio.mmi culprit.fa > culprit.fa.sam`

The std error from minimap2 was:

```text
[M::main::9.683*1.00] loaded/built the index for 195 target sequence(s)
[M::mm_mapopt_update::13.503*0.99] mid_occ = 470
[M::mm_idx_stat] kmer size: 19; skip: 10; is_HPC: 1; #seq: 195
[M::mm_idx_stat::14.957*0.99] distinct minimizers: 86648969 (45.01% are singletons); average occurrences: 4.325; average spacing: 8.271
/tmp/slurmd/job39041/slurm_script: line 4: 136713 Segmentation fault      (core dumped) minimap2 -t 1 -L -ax map-pb hg38_no_alt.fasta.pacbio.mmi culprit.fa > culprit.fa.sam
```

The hg38 reference I used can be downloaded from NCBI at ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz

After uncompressing the ref fasta file, an index was created before mapping with command `minimap2 -d hg38_no_alt.fasta.pacbio.mmi -x map-pb hg38_no_alt.fasta`

minimap2 version: 2.3-r536-dirty

**Comment by lh3** [comment]
Thanks very much. This is exactly the example I would like to see. Extremely helpful.

It appears that this bug might be lurking in the earliest piece of code of minimap2. It is tricky to fix. Stay tuned.

**Comment by lh3** [comment]
Fixed via f22a94e. Thanks again. Really appreciate.
